user_model.ml

```diff
@@ -101,13 +161,84 @@ let generate_token ?(expires_in = 3600) () =
   let token = generate_uuid () in
   { token_type = "Bearer"; value = Uuidm.to_string token; expires_in }
 
+let generate_cookie ~name ~uuid ?(expires_in = 3600) () =
+  let id = generate_uuid () in
+  {
+    name;
+    value = Base64.encode_string (Uuidm.to_string id);
+    expires_in;
+    uuid = Some uuid;
+  }
+
+let create_user_session_map (users : user list) : (string * user) list =
+  List.map
+    (fun user ->
+      let cookie =
+        List.find
+          (fun (cookie : cookie) -> String.equal cookie.name "molly_session")
+          user.cookies
+      in
+      (cookie.value, user))
+    users
+
+let create_user_uuid_map (users : user list) : (string * user) list =
+  List.map (fun user -> (user.uuid, user)) users
+
+let find_user_by_key (uuid : string) (user_map : (string * user) list) :
+    user option =
+  List.assoc_opt uuid user_map
+
 let create_user ~name ~email ~password =
   let uuid = Uuidm.to_string (generate_uuid ()) in
   let password = hash_password password uuid in
+  let auth_token = generate_token () in
+  let session =
+    generate_cookie ~name:"molly_session" ~expires_in:604800 ~uuid ()
```

Comment:
sonce this 604800 is used in several places, let's introduce a top-level binding `let week = 604800 (* a week = 7 days * 24 hours * 60 minutes * 60 seconds *)` and use that